scheduler/README.md

```diff
@@ -0,0 +1,196 @@
+# Nomad Scheduler
+
+This package holds the logic behind Nomad schedulers. The `Scheduler` interface
+is implemented by two objects:
+
+- `GenericScheduler` and
+- `SystemScheduler`.
+
+The `CoreScheduler` object also implements this interface, but it's use is
+purely internal, the core scheduler does not schedule any user jobs.
+
+Nomad scheduler's task is to, given an evaluation, produce a plan of placing the
+desired allocations on feasibile nodes. Consult [Nomad documentation][0] for
+more details.
+
+The diagram below illustrates this process for the service and system schedulers
+in more detail:
+
+```
+                                 +--------------+        +-----------+                                               +-------------+      +----------+
+                                 |   cluster    |        |feasibility|                          +-------------+      |    score    |      |   plan   |
+    Service and batch jobs:      |reconciliation|------->|   check   |------------------------->|     fit     |----->| assignment  |----->|submission|
+                                 +--------------+        +-----------+                          +-------------+      +-------------+      +----------+
+
+- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+                                                         +-----------+     +--------------+                                               +----------+
+ System and sysbatch jobs:                               |feasibility|     |     node     |     +-------------+                           |   plan   |
+                                                         |   check   |---->|reconciliation|---->|     fit     |-------------------------->|submission|
+                                                         +-----------+     +--------------+     +-------------+                           +----------+
```

Comment:
This ordering seems to come from the Reconciler Revisited proposal but taking a second look at the system scheduler I'm not so sure. The `computeJobAllocs` method calls the reconciler-like `diffSystemAllocs` before it calls `computePlacements` (which selects from the stack).

I like that we're differentiating this from "cluster reconciliation", but I think we want to move it one step earlier?

Reply:
oh interesting. I saw https://github.com/hashicorp/nomad/blob/main/scheduler/scheduler_system.go#L157-L160 and then https://github.com/hashicorp/nomad/blob/main/scheduler/scheduler_system.go#L163-L166 happening after, but we're just constructing the stack and not making a selection before we "reconcile." Good catch, I'll update the diagram. 

No misalignment anymore btw?

Reply:
21373ea